tests/lib/unit/API/Repository/Value/Filter/FilterTest.php

```diff
@@ -0,0 +1,323 @@
+            $filter->reset();
+            self::assertEmpty($filter->getCriterion());
+            self::assertEmpty($filter->getSortClauses());
+        }
+    }
+
+    /**
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::getCriterion
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::getSortClauses
+     *
+     * @dataProvider getComplexFilterTestData
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\SortClause[] $expectedSortClauses
+     */
+    public function testBuildingComplexFilter(
+        Filter $filter,
+        Criterion $expectedCriterion,
+        array $expectedSortClauses
+    ): void {
+        self::assertEquals($expectedCriterion, $filter->getCriterion());
+        self::assertEquals($expectedSortClauses, $filter->getSortClauses());
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function getComplexFilterTestData(): iterable
+    {
+        $parent1Criterion = new Criterion\ParentLocationId(1);
+        $engGBCriterion = new Criterion\LanguageCode('eng-GB');
+        $parent2Criterion = new Criterion\ParentLocationId(2);
+        $gerDECriterion = new Criterion\LanguageCode('ger-DE');
+
+        $criterion = new Criterion\LogicalOr(
+            [
+                new Criterion\LogicalAnd(
+                    [
+                        $parent1Criterion,
+                        $engGBCriterion,
+                    ]
+                ),
+                new Criterion\LogicalAnd(
+                    [
+                        $parent2Criterion,
+                        $gerDECriterion,
+                    ]
+                ),
+            ]
+        );
+        $sortClauses = [
+            new SortClause\Location\Priority(),
+            new SortClause\ContentName(Query::SORT_DESC),
+        ];
+        $filter = new Filter();
+        $filter
+            ->withCriterion($criterion->criteria[0])
+            ->orWithCriterion($criterion->criteria[1])
+            ->withSortClause($sortClauses[0])
+            ->withSortClause($sortClauses[1]);
+
+        yield 'parent1 AND language1 OR parent2 AND language2' => [
+            $filter,
+            $criterion,
+            $sortClauses,
+        ];
+
+        // ------------------------------------------------------------------------------------
+
+        $criterion = new Criterion\LogicalAnd(
+            [
+                new Criterion\LogicalOr(
+                    [
+                        new Criterion\LogicalAnd(
+                            [
+                                $parent1Criterion,
+                                $engGBCriterion,
+                            ]
+                        ),
+                        $parent2Criterion,
+                    ]
+                ),
+                $gerDECriterion,
+            ]
+        );
+
+        $filter = clone $filter;
+        $filter
+            ->reset()
```

Comment:
This lines could be simpliied to "new Filter()"

Reply:
> This lines could be simpliied to "new Filter()"

AFAIR the point here was to test resetting after cloning.